### ISSUE TITLE
Make swiper usable with a custom candidates list

### ISSUE THREAD

**Issue by rgrinberg** [abo-abo/swiper#1426]
I really like swiper's user interface and would like to make it usable in a few other places. For example, I'd like to be able to use swiper to navigate my flycheck errors. I was able to hack something together by modifying `swiper--candidates` to return candidates corresponding to line numbers and then calling `swiper--ivy`. But perhaps there should be an easier way of accomplishing this. I'd like to have this available for xref candidates as well for example.

**Comment by abo-abo** [comment]
If you know how to collect a set of strings and positions, the rest is as easy as passing it to `ivy-read`:

```elisp
(defun example ()
  (interactive)
  (let ((cands '(("one" . 1)
                 ("two" . 2))))
    (ivy-read "goto: " cands
              :action (lambda (x) (goto-line (cdr x))))))
```

I don't use flycheck, so I can't help you further with the collection code.

**Comment by rgrinberg** [comment]
Yeah, but this isn't as nice as using swiper. Which lets me navigate quickly with `C-n` and `C-p`, syntax coloring, line numbers, etc.
